pkg/core/app/hooks.go

```diff
@@ -0,0 +1,29 @@
+package app
+
+import (
+	"go.keploy.io/server/v2/pkg/platform/docker"
+	"go.uber.org/zap"
+)
+
+// AppRuntimeHooks defines extension points used during app lifecycle.
+// Implementations may mutate compose spec in-place.
+type AppRuntimeHooks interface {
+	BeforeDockerComposeSetup(logger *zap.Logger, compose *docker.Compose, serviceName string) (bool, error)
+	BeforeDockerSetup(logger *zap.Logger, cmd string) (string, error)
+}
+
+// RuntimeHooks is the singleton used by runtime; can be overridden by other builds.
+var RuntimeHooks AppRuntimeHooks = defaultAppRuntimeHooks{}
```

Comment:
Rather than initialising here, we can do it at this [level](https://github.com/keploy/keploy/blob/688e35433c3e695af75854d018cffa698b906548/cli/provider/core_service_linux.go#L71). And it would be like
```go
 app.HooksImpl = app.NewHooks(logger)
 ```